### ISSUE TITLE
Regression: smooth(4) shows “smooth() can only be used inside settings()” error

### ISSUE THREAD

**Issue by benfry** [benfry/processing4#149]
Looks like `smooth()` not properly being hoisted out of `setup()` in 4.x, inside the PDE this results in “smooth() can only be used inside settings()” in the console.

```processing
void setup(){ 
  size(300,300, P2D);
  smooth(4);
}

void draw(){
  background(0);
  fill(255,0,0);
  ellipse(100,100,100,100);
  fill(0,255,0);
  ellipse(150,150,100,100);
}
```

**Comment by sampottinger** [comment]
Weird... this was resolved at one point but I’ll take a look.

**Comment by benfry** [comment]
Any luck?

**Comment by sampottinger** [comment]
Hey there!

Sorry this got a little buried. If I can’t get to this over the weekend, I’ll try over the holiday next week.

Thanks,
Sam

**Comment by sampottinger** [comment]
I have a PR for you @benfry at #152. Apologies for the delay!

<img width="900" alt="Screen Shot 2020-11-21 at 11 59 19 AM" src="https://user-images.githubusercontent.com/110391/99886555-4edb3180-2bf2-11eb-9e03-8269b149b389.png">

**Comment by REAS** [comment]
this also seems to affect pixelDensity(), same error report. it's my guess the fix is the same, but I'll follow up after ben accepts the pr.

**Comment by benfry** [comment]
Resolved for 4.0a3 with code from @sampottinger.
